### DISCUSSION TITLE
New Bing开始采用新的接口鉴权方式

### DISCUSSION THREAD

**Discussion by juzeon** [viopsa233/AI-testers-Forum#99] [技术交流]
今天发现New Bing采用了新的接口鉴权方式。

以往使用已登录账号的cookie访问`/turing/conversation/create`的返回结果：

```json
{
    "conversationId": "51D|BingProd|85AA3DE6B031DE6EEF2B9EEF0656450747D207F367A155FCA69C21D54EBCC737",
    "clientId": "22B7EBD1226F63D516F5F98223236252",
    "conversationSignature": "2Uc1/b3I8tbn2RExJN1S8Bb69eYtj0n4NTYaN/9QLiw=",
    "result": {
        "value": "Success",
        "message": null
    }
}
```

包含一个`conversationSignature`字段，该字段需要在后面使用wss的接口进行交互时发送。

注意`conversationId`中，如果是未登录用户，显示的是`BingProdUnAuthenticatedUsers`；如果是已登录用户，显示的是`BingProd`。

如今未登录用户访问这个接口仍然包括`conversationSignature`字段，而已登录用户访问则不包括这个字段了。

取而代之的是，在HTTP Response Header中会返回一个头：`X-Sydney-Encryptedconversationsignature`。

而调用websocket接口的时候原来是直接调用：`wss://sydney.bing.com/sydney/ChatHub`。

现在需要加一个query参数：`wss://sydney.bing.com/sydney/ChatHub?sec_access_token=xxxxxxxxxxxxxxxxx`。

其中`xxxxxxxxxxxxxxxxx`就是刚刚获取到的`X-Sydney-Encryptedconversationsignature`的值，注意需要url编码。

SydneyQt已修复这个问题：https://github.com/juzeon/SydneyQt/commit/d633bde63f46399aac4a9081ef3260c8b25dcab4

更新：发现它的接口有时候用的是旧版的方式，有时候用的又是新版的方式？？https://github.com/juzeon/SydneyQt/commit/b93a802918734ceabaf83a270e256aa5c9f8666f

**Comment by Harry-zklcdc** [comment]
目前已知在两个礼拜前，就有这个迹象了，不过一直反反复复，一直启用停用
[`go-proxy-bingai`](https://github.com/Harry-zklcdc/go-proxy-bingai) 这个项目我也给他修复了

不过目前最令人激动的是文件上传的功能，已经找到对应的入口，不过还缺少关键参数，导致无法调用
![8dd378668c5f4b909fa9f7ec8b7cf0a7](https://github.com/viopsa233/AI-testers-Forum/assets/21104213/0be669e8-4138-44d5-bc68-ba50a2b12a70)
